pkg/arm/data/master-startup.sh

```diff
@@ -57,602 +57,41 @@ done
-    volumeMounts:
-    - mountPath: /etc/etcd
-      name: etcd-config
-      readOnly: true
-    - mountPath: /var/lib/etcd
-      name: etcd-data
-    workingDir: /var/lib/etcd
-  hostNetwork: true
-  dnsPolicy: "None"
-  dnsConfig:
-    nameservers:
-      - 168.63.129.16
-    searches:
-      - $(hostname -d)
-  priorityClassName: system-node-critical
-  volumes:
-  - hostPath:
-      path: /etc/etcd
-    name: etcd-config
-  - hostPath:
-      path: /var/lib/etcd
-    name: etcd-data
+cat >/etc/origin/cloudprovider/configstoragekey.json <<'EOF'
+{{ .Extra.ConfigStorageKey }}
 EOF
-
-cat >/etc/origin/node/pods/apiserver.yaml <<'EOF'
-apiVersion: v1
-kind: Pod
-metadata:
-  annotations:
-    scheduler.alpha.kubernetes.io/critical-pod: ''
-  labels:
-    openshift.io/component: api
-    openshift.io/control-plane: 'true'
-  name: master-api
-  namespace: kube-system
-spec:
-  containers:
-  - args:
-    - start
-    - master
-    - api
-    - --config=/etc/origin/master/master-config.yaml
-{{- if $.Extra.TestConfig.RunningUnderTest }}
-    - --loglevel=4
-{{- else }}
-    - --loglevel=2
-{{- end }}
-    command:
-    - openshift
-    image: {{ .Config.Images.ControlPlane | quote }}
-    imagePullPolicy: Always
-    livenessProbe:
-      httpGet:
-        path: healthz
-        port: 443
-        scheme: HTTPS
-      initialDelaySeconds: 45
-      timeoutSeconds: 10
-    name: api
-    readinessProbe:
-      httpGet:
-        path: healthz/ready
-        port: 443
-        scheme: HTTPS
-      initialDelaySeconds: 10
-      timeoutSeconds: 10
-    securityContext:
-      privileged: true
-    volumeMounts:
-    - mountPath: /etc/origin/master
-      name: master-config
-      readOnly: true
-    - mountPath: /etc/origin/cloudprovider
-      name: master-cloud-provider
-      readOnly: true
-    - mountPath: /var/lib/origin/
-      name: master-data
-    - mountPath: /etc/pki
-      name: master-pki
-  hostNetwork: true
-  priorityClassName: system-node-critical
-  volumes:
-  - hostPath:
-      path: /etc/origin/master
-    name: master-config
-  - hostPath:
-      path: /etc/origin/cloudprovider
-    name: master-cloud-provider
-  - hostPath:
-      path: /var/lib/origin
-    name: master-data
-  - hostPath:
-      path: /etc/pki
-    name: master-pki
+cat >/etc/origin/cloudprovider/registrystoragekey.json <<'EOF'
+{{ .Extra.RegistryStorageKey }}
 EOF
 
-cat >/etc/origin/node/pods/controllers.yaml <<'EOF'
-apiVersion: v1
-kind: Pod
-metadata:
-  name: controllers
-  namespace: kube-system
-  labels:
-    openshift.io/component: controllers
-    openshift.io/control-plane: "true"
-spec:
-  containers:
-  - args:
-    - start
-    - master
-    - controllers
-    - --config=/etc/origin/master/master-config.yaml
-    - --listen=https://0.0.0.0:8444
+######################################################
+# run startup container
+logger -t master-startup.sh "running startup container"
+while ! docker pull quay.io/asalkeld/aos-init; do
```

Comment:
some wiring is missing - this should honour STARTUP_IMAGE

Reply:
yip done